tempodb/compactor.go

```diff
@@ -380,3 +422,21 @@ func (i instrumentedObjectCombiner) Combine(dataEncoding string, objs ...[]byte)
 	}
 	return b, wasCombined, err
 }
+
+// doForAtLeast executes the function f. It blocks for at least the passed duration but can go longer. if context is cancelled after
+// the function is done we will bail immediately. in the current use case this means that the process is shutting down
+// we don't force f() to cancel, we assume it also responds to the cancelled context
+func doForAtLeast(ctx context.Context, dur time.Duration, f func()) {
+	startTime := time.Now()
+	f()
+	elapsed := time.Since(startTime)
+
+	if elapsed < dur {
+		ticker := time.NewTicker(dur - elapsed)
+
+		select {
+		case <-ticker.C:
+		case <-ctx.Done():
```

Comment:
Return here?

Reply:
both cases will just fall through to the end of the function. this basically just blocks until one of the two things occurs.

Reply:
Yeah, its interesting but no issues.